File: asyncawait-ts/basics03.ts
import {sleep} from "./lib";

function main() {
    (async () => {
        await doWorld();
    })();
    // cannot block
}

async function doWorld() {
    const p = (async () => {
        await sleep(1000);
        console.log('World!');
    })();
    console.log('Hello');
    await p;
}

main();

File: coroutine-py/basics03.py
import asyncio


def main():
    asyncio.run(async_main())


async def async_main():
    await do_world()


async def do_world():
    async with asyncio.TaskGroup() as tg:
        tg.create_task(do_world_do_world())
        print('Hello')


async def do_world_do_world():
    await asyncio.sleep(1)
    print('World!')


if __name__ == "__main__":
    main()
